### PR TITLE
Add sts_endpoint parameter

### PR DESCRIPTION
**Pull request by PettitWesley**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -125,15 +127,11 @@ func (config OutputPluginConfig) Validate() error {
 
 // NewOutputPlugin creates a OutputPlugin object
 func NewOutputPlugin(config OutputPluginConfig) (*OutputPlugin, error) {
-	sess, err := session.NewSession(&aws.Config{
-		Region: aws.String(config.Region),
-	})
+	client, err := newCloudWatchLogsClient(config.Region, config.RoleARN, config.CWEndpoint, config.STSEndpoint, config.CredsEndpoint, config.LogFormat)
```
Comment (sonofachamp) [inline cloudwatch/cloudwatch.go]: Should we just pass the `config` in?

File: cloudwatch/cloudwatch.go [hunk 2]
```diff
@@ -102,6 +103,7 @@ type OutputPluginConfig struct {
 	RoleARN          string
 	AutoCreateGroup  bool
 	CWEndpoint       string
+	STSEndpoint      string
```
Comment (sonofachamp) [inline cloudwatch/cloudwatch.go:106]: I noticed we don't have an `OutputPluginConfig` struct in our other plugins. I think we should be consistent across the plugins, so maybe we should add items to the other plugin repos to adopt this pattern?

File: cloudwatch/cloudwatch.go [hunk 3]
```diff
@@ -157,40 +155,52 @@ func NewOutputPlugin(config OutputPluginConfig) (*OutputPlugin, error) {
...
-		defaultResolver := endpoints.DefaultResolver()
-		cwCustomResolverFn := func(service, region string, optFns ...func(*endpoints.Options)) (endpoints.ResolvedEndpoint, error) {
-			if service == "logs" {
-				return endpoints.ResolvedEndpoint{
-					URL: endpoint,
-				}, nil
-			}
-			return defaultResolver.EndpointFor(service, region, optFns...)
+func newCloudWatchLogsClient(region, roleARN, cwEndpoint, stsEndpoint, credsEndpoint, logFormat string) (*cloudwatchlogs.CloudWatchLogs, error) {
+	defaultResolver := endpoints.DefaultResolver()
```
Comment (sonofachamp) [inline cloudwatch/cloudwatch.go]: Can we move this into the `customResolverFn` function definition instead of referencing it via a closure?

**Review by sonofachamp — APPROVED** [review]

File: cloudwatch/cloudwatch.go (hunk 2 above)
Comment (PettitWesley) [reply cloudwatch/cloudwatch.go:106]: We should, the config struct is a lot cleaner. I don't want to do that refactoring right now though...

File: cloudwatch/cloudwatch.go (hunk 1 above)
Comment (PettitWesley) [reply cloudwatch/cloudwatch.go]: 👍

**Pull request merged**